test/metadata_tests.cpp

```diff
@@ -84,6 +85,30 @@ BOOST_AUTO_TEST_CASE( head_bucket ) {
     free_client(client);
 }
 
+BOOST_AUTO_TEST_CASE( head_folder ) {
+    printf("-----Testing head_folder-------\n");
+    ds3_error* error;
+    ds3_client* client = get_client();
+    const char* bucket_name = "head_folder_test";
+    ds3_request* request = ds3_init_put_bucket(bucket_name);
+    error = ds3_put_bucket(client, request);
+    ds3_free_request(request);
+    handle_error(error);
+
+    const char* test_folder = "test_folder/";
+    request = ds3_init_put_object_for_job(bucket_name, test_folder, 0, 0, NULL);
+    error   = ds3_put_object(client, request, NULL, NULL);
+    ds3_free_request(request);
+    handle_error(error);
+
+    request = ds3_init_head_bucket(bucket_name);
+    error = ds3_head_bucket(client, request);
```

Comment:
Err, yes.  CopyPasta error.

with that change, this doesn't work... and will require a fundamental re-think.

-----Testing head_folder-------
ds3_error_message: The object name parameter is required.
test.cpp(58): fatal error in "head_folder": Test failed with a ds3_error